### ISSUE TITLE
Segmentation fault (core dumped) in `tf.data.experimental.service.DispatchServer`

### ISSUE THREAD

**Issue by x0w3n** [tensorflow/tensorflow#76720]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.18.0-dev20240925

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04.3 LTS (x86_64)

### Python version

3.9.13

### Current behavior?

tf.data.experimental.service.DispatchServer triggered a crash, when the start parameter is false.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

def test():
    tf.data.experimental.service.DispatchServer(start=False)
test()
```

### Relevant log output

```shell
2024-09-28 20:20:16.983171: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-09-28 20:20:17.013139: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-09-28 20:20:17.063675: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-09-28 20:20:17.087886: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-09-28 20:20:17.147364: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 AVX512F AVX512_VNNI AVX512_BF16 AVX512_FP16 AVX_VNNI AMX_TILE AMX_INT8 AMX_BF16 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
Segmentation fault (core dumped)
```

**Comment by Venkat6871** [comment]
Hi **@x0w3n** ,
I tried running your code on Colab using TensorFlow v2.17.0 & the nightly version and faced the same issue. I also tried an alternative approach. If we start the server manually, then it works fine. Here, I am providing the [gist](https://colab.sandbox.google.com/gist/Venkat6871/07d0c607c5221bc63813d6086604f37b/76720_tf-2-17-0-nightly-v.ipynb) for your reference. I hope it will be useful for you.
Thank you!
